Haskell/Blarney/Backend/Utils.hs

```diff
@@ -0,0 +1,58 @@
+{-|
+Module      : Blarney.Backend.Utils
```

Comment:
`Backend.Utils` --> `Backend.ModuleHierarchy` ?

Reply:
Do you mean actually pull this out of `Blarney.Backend`? I am not sure if I am too happy about it living straight under `Blarney`. Maybe it belongs in `Blarney.Netlist`?

Reply:
Was just wondering if we want a more specific name than Utils, that's all.

Reply:
Ah! sorry, I misread the first comment... Yeah I don't mind. We can change it for something more specific. But now that I went off on a tangent, I am actually wondering whether this is where this should live in the first place? Should it be in `Core`?

Reply:
Yeah, the higher the better I guess